### PR TITLE
ci: Raise Documentation Generation Errors

### PR DESCRIPTION
**Pull request by fpetrini15**
#### What does the PR do?
Raise errors in the documentation generation script so that CI doesn't falsely report that the job succeeded

#### Checklist
- [X] PR title reflects the change and is of format `<commit_type>: <Title>`
- [X] Changes are described in the pull request.
- [X] Related issues are referenced.
- [X] Populated [github labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels) field
- [ ] Added [test plan](#test-plan) and verified test passes.
- [X] Verified that the PR passes existing CI.
- [X] Verified copyright is correct on all changed files.
- [ ] Added _succinct_ git squash message before merging [ref](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
- [ ] All template sections are filled out.

#### Commit Type:
Check the [conventional commit type](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type)
box here and add the label to the github PR.
- [ ] build
- [X] ci
- [ ] docs
- [ ] feat
- [ ] fix
- [ ] perf
- [ ] refactor
- [ ] revert
- [ ] style
- [ ] test

### CODE REVIEW

**Review by mc-nv — APPROVED** [review]

File: docs/generate_docs.py [hunk 1]
```diff
@@ -155,7 +153,7 @@ def clone_from_github(repo, tag, org):
         subprocess.run(clone_command, check=True)
         log_message(f"Successfully cloned {repo}")
     except subprocess.CalledProcessError as e:
-        log_message(f"Failed to clone {repo}. Error: {e}")
+        raise (e)
```
Comment (yinggeh) [inline docs/generate_docs.py:156]: Does `e` already include the repo's info?

Comment (fpetrini15) [reply docs/generate_docs.py:156]: Yes, the exception in the job I linked looks like this: 
```bash
subprocess.CalledProcessError: Command '['git', 'clone', '--branch', '', '--single-branch', 'https://github.com/triton-inference-server/perf_analyzer.git']' returned non-zero exit status 128.
```
This should be sufficient to understand why the script failed.

File: docs/generate_docs.py [hunk 2]
```diff
@@ -123,9 +123,7 @@ def run_command(command):
             stderr=subprocess.PIPE,
         )
     except subprocess.CalledProcessError as e:
-        log_message(f"Error executing command: {e.cmd}")
-        log_message(e.output)
-        log_message(e.stderr)
+        raise (e)
 
 
 def clone_from_github(repo, tag, org):
```
Comment (yinggeh) [inline docs/generate_docs.py:129]: Can you also remove `print(repo_url)`? I think it was for debugging purpose.

**Review by yinggeh — CHANGES_REQUESTED** [review]

**Comment by yinggeh** [comment]
In `get_git_repo_name`, catch statement should raise error instead of `return None`. Can you also fix that?

**Review by yinggeh — APPROVED** [review]

**Pull request merged**